### PR TITLE
build: Add basic CI test pipelines

### LINKED ISSUE #7: Create CI/CD pipelines
We need to create some basic CI/CD pipelines to run through tests to at least run through tests in the repo itself.

### PR DESCRIPTION
**Pull request by sunchao**
This adds some basic CI test pipelines for the project. Basically run tests in the Rust and Java/Scala side within the repo.

Closes #7

### CODE REVIEW

**Comment by sunchao** [comment]
Need to find a way to test this. Also this only run tests in Linux for Spark 3.4 atm, and we need more combinations.

**Comment by sunchao** [comment]
@viirya @andygrove this is ready now. Tested [here](https://github.com/sunchao/arrow-datafusion-comet/actions/runs/7896769061).

File: core/src/errors.rs [hunk 1]
```diff
@@ -651,6 +651,7 @@ mod tests {
     /// See [`object_panic_exception`] for a test which involves generating a panic and verifying
     /// that the resulting stack trace includes the offending call.
     #[test]
+    #[ignore]
```
Comment (sunchao) [inline core/src/errors.rs:654]: the golden file for this test is not added yet, so ignore for now

File: spark/src/test/scala/org/apache/spark/sql/comet/CometPlanStabilitySuite.scala [hunk 2]
```diff
@@ -297,7 +297,7 @@ class CometTPCDSV1_4_PlanStabilitySuite extends CometPlanStabilitySuite {
     new File(baseResourcePath, "approved-plans-v1_4").getAbsolutePath
 
   tpcdsQueries.foreach { q =>
-    test(s"check simplified (tpcds-v1.4/$q)") {
+    ignore(s"check simplified (tpcds-v1.4/$q)") {
```
Comment (sunchao) [inline spark/src/test/scala/org/apache/spark/sql/comet/CometPlanStabilitySuite.scala:300]: depends on #10

File: .github/workflows/pr_build.yml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      - name: Build common module (pre-requisite for Rust tests)
+        run: |
+          cd common
+          ../mvnw clean compile -DskipTests
+
+      - name: Run cargo test
+        run: |
+          cd core
+          # This is required to run some JNI related tests on the Rust side
+          LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$JAVA_HOME/lib:$JAVA_HOME/lib/server cargo test
```
Comment (viirya) [inline .github/workflows/pr_build.yml:82]: RUST_BACKTRACE=1?

File: .github/workflows/pr_build.yml [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        run: |
+          echo "JAVA_HOME=/usr/lib/jvm/java-${{env.JAVA_VERSION}}-openjdk-amd64" >> $GITHUB_ENV
+
+      - name: Run Maven compile
+        run: |
+          ./mvnw compile test-compile scalafix:scalafix -Psemanticdb
+
+      - name: Run tests
+        run: |
+          SPARK_HOME=`pwd` ./mvnw clean install
```
Comment (viirya) [inline .github/workflows/pr_build.yml:112]: Hmm, we use `./mvnw verify` in Makefile, do we need to do that?

Comment (viirya) [reply .github/workflows/pr_build.yml:112]: Don't we need `BOSON_CONF_DIR` and `SPARK_HOME`?

**Review by viirya — APPROVED** [review]

**Comment by viirya** [comment]
There are a log message like this:

```
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-utils/3.0.22/plexus-utils-3.0.22.jar (245 kB at 2.3 MB/s)
Progress (1): 1.0/2.3 MB
Progress (1): 1.1/2.3 MB
Progress (1): 1.1/2.3 MB
...
```

It is really distracting for looking at the logs. Can we silent it?

File: .github/workflows/pr_build.yml (hunk 4 above)
Comment (sunchao) [reply .github/workflows/pr_build.yml:112]: `install` covers `verify`: https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html#a-build-lifecycle-is-made-up-of-phases

`COMET_CONF_DIR` is optional and by default, Rust test outputs is directed to stdout.

**Comment by sunchao** [comment]
> It is really distracting for looking at the logs. Can we silent it?

Hmm perhaps we can cache the Maven dependencies so these will only show up in the first time. Let me check

File: .github/workflows/pr_build.yml (hunk 3 above)
Comment (sunchao) [reply .github/workflows/pr_build.yml]: yea added

**Review by andygrove — APPROVED** [review]

**Comment by sunchao** [comment]
Thanks! merged. Let me check if the Maven cache did work. I'll create followups if it doesn't.

**Pull request merged**